analyses/tmb-compare/compare-tmb-calculations.Rmd

```diff
@@ -0,0 +1,198 @@
+- [How different are the mutation counts?](#how-different-are-the-mutation-counts)
+- [Calculate percent difference in BED size](#calculate-percent-difference-in-bed-size)
+- [Session Info](#session-info)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/tmb-compare/compare-tmb-calculations.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+```{r}
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Declare names of input and output directories.
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+scratch_dir <- file.path("..", "..", "scratch")
+```
+
+Custom function for plotting the TMB. 
+
+```{r}
+source(file.path("util", "cdf-plot-function.R"))
+```
+
+## Set up consensus data
+
+Read in the three consensus TMB files. 
+
+This is the main TMB that is used in other downstream results. 
+
+```{r}
+tmb_ccdl_maftools <- data.table::fread(file.path(
+  "..",
+  "snv-callers",
+  "results",
+  "consensus",
+  "pbta-snv-mutation-tmb-coding.tsv"
+))
+```
+
+This is the FoCR-defined filter TMB results. 
+
+```{r}
+tmb_ccdl_focr <- data.table::fread(file.path(
+  "..",
+  "snv-callers",
+  "results",
+  "focr_filter",
+  "pbta-snv-mutation-tmb-coding.tsv"
+))
+```
+
+This is the d3b calculated results. 
+
+```{r}
+tmb_d3b <- data.table::fread(file.path(
+  "TMB_d3b_code",
+  "outputs",
+  "pbta-snv-consensus-TMB_intarget.txt"
+))
+```
+
+Combine the three datasets into one data.frame. 
+
+```{r}
+all_tmb <- tmb_ccdl_maftools %>% 
+  dplyr::inner_join(tmb_ccdl_focr, by = "Tumor_Sample_Barcode", 
+                    suffix = c("_maftools", "_focr")) %>% 
+  dplyr::inner_join(tmb_d3b, by = "Tumor_Sample_Barcode") %>% 
+  dplyr::rename(tmb_d3b = TMB, 
+                mutation_count_d3b = count)
+```
+
+## Percent Differences in TMB
+
+```{r}
+all_tmb <- all_tmb %>% 
+  dplyr::mutate(perc_diff_focr = abs(tmb_focr - tmb_d3b) / max(tmb_focr, tmb_d3b), 
+                perc_diff_maf = abs(tmb_maftools - tmb_d3b) / max(tmb_maftools, tmb_d3b))
+```
+
+This is the same regardless of which definition (maftools/FoCR) is used. 
+
+```{r}
+all_tmb %>% 
+  dplyr::filter(perc_diff_focr > .05)
+
+all_tmb %>% 
+  dplyr::filter(perc_diff_maf > .05)
+```
+
+Most are HGAT samples, but probably because they are outliers. 
+
+## How different are the mutation counts?
+
+Select the counts columns only. 
+
+```{r}
+counts_only <- all_tmb %>% 
+  dplyr::select(mutation_count_focr, mutation_count_maftools, mutation_count_d3b)
+
+# Print this out
+counts_only
+```
+
+Plot this. 
+
+```{r}
+counts_only %>%
+  GGally::ggpairs(mapping = ggplot2::aes(alpha = 0.05)) +
+  ggplot2::theme_classic()
+```
+
+Counts look *mostly* the same, but those outliers make the plots not super useful. 
```

Comment:
I might try plotting `log1p()` values to make this more comparable. Adding a y=x line will also be informative, showing that d3b counts are always lower.